### ISSUE TITLE
Backend eliminates side effect in `{???; Int}.box(42)`

### ISSUE THREAD

**Issue by retronym** [scala/scala-dev#85]

**Comment by sjrd** [comment]
Yep ... So does the Scala.js backend http://www.scala-js-fiddle.com/gist/6e1f912c253e13e69de7
Bug-compatibility FTW!

**Comment by lrytz** [comment]
Related to https://issues.scala-lang.org/browse/SI-5304. wip https://github.com/scala/scala/compare/2.12.x...lrytz:sd85?expand=1 has issues
  - doesn't treat `{(); Int}.box(1)` as pure because of how `isExprSafeToInline` implements `case Block`
  - treats `{val x = O; Int}.box(1)` as pure (where `O` is a module that may have side-effects in its initializer), again due to the impl of `isExprSafeToInline`

**Comment by adriaanm** [comment]
Can we push to RC1 (or 2.12.x?)

**Comment by lrytz** [comment]
yes, this is not a regression anyway
